tests/frontend/test_circuitspecs_chip2.py

```diff
@@ -367,6 +395,62 @@ def test_no_unitary(self, tol):
 
         assert program_equivalence(res, expected, atol=tol)
 
+        # double check that the applied symplectic is correct
+
+        # remove the Fock measurements
+        res.circuit = res.circuit[:-1]
+
+        # extract the Gaussian symplectic matrix
+        O = res.compile("gaussian_unitary").circuit[0].op.p[0]
+
+        # construct the expected symplectic matrix corresponding
+        # to just the initial two mode squeeze gates
+        S = TMS(SQ_AMPLITUDE, 0)
+
+        expected = np.zeros([2*8, 2*8])
+        idx = np.arange(2*8).reshape(4, 4).T
```

Comment:
What's the purpose of the transpose here? :thinking:

Reply:
Without the transpose, the S2gates would be applied to modes in a 'cascade', i.e., `[0, 1]`, `[2, 3]`, `[4, 5]` etc.

With the transpose, the cascade now properly couples between the modes: `[0, 4]`, `[1, 5]`, `[2, 6]` etc.